examples/get-media-details/main.py

```diff
@@ -0,0 +1,73 @@
+import json
+
+from utils.utils import PublicAPIClient, get_standard_arguments
+
+
+def main():
+    args = get_standard_arguments(
+        [
+            (["media_id"], {"type": int, "help": "the media to get details about"}),
+        ]
+    )
+    media_id = args.media_id
+
+    public_api_client = PublicAPIClient(args.subdomain)
+    media = get_media_details(public_api_client, media_id)
+    captions = get_media_details(public_api_client, media_id, "caption_files")
+    courses = get_media_details(public_api_client, media_id, "courses")
+    tags = get_media_details(public_api_client, media_id, "tags")
+    users = get_media_details(public_api_client, media_id, "users", "user_permissions")
+
+    response = get_subdict(
```

Comment:
Does this filter out the unnecessary data? something like serialization?

Reply:
If we don't select the required attributes, we will return thumbnail_url and transcoding_status as well. This *can* be okay, but the ticket explicitly details the response, and they are not included. This is an example script, you we can either say that simple is better (and just return everything) or say that it's good to show them how they can only get what's interesting for them. IDK. WDYT @zoltan-sebestyen-instructure ?

Reply:
Oh and I think I didn't quite answer your original questions. So yes, it's to filter out unnecessary data. And no, serialization is something totally different: that converts an object (e.g. database model) into a string (typically formatted as JSON, but it can be anything).

This is simply a function to only return a subset of a dict:
```
def get_subdict(dict_, keys):
    return {key: dict_[key] for key in keys}
```

It creates and returns a new dict from the old `dict_`, selecting only keys that are specified in `keys`.

The `{something for something in something}` construct is called "dictionary comprehension" if you wanna google it, it's pretty standard in Python.

Reply:
Also: the reason I named the first arguement `dict_` is that it's a convention to add a trailing underscore to a name that is a keyword/reserved. E.g. if you have a function that gets a list argument and you have no better name than "list", use `list_`. If there's a more semantical name, of course use that. But since this is general a dictionary manipulation function, `dict_` is probably the best name.

Reply:
@cu2 
> This is an example script, so we can either say that simple is better (and just return everything) or say that it's good to show them how they can only get what's interesting for them

makes sense to me.

Reply:
or MS2M